lib/archethic/contracts/loader.ex

```diff
@@ -45,24 +45,66 @@ defmodule Archethic.Contracts.Loader do
   """
   @spec load_transaction(Transaction.t(), list()) :: :ok
   def load_transaction(tx, opts \\ []) do
-    from_db = Keyword.get(opts, :from_db, false)
-    from_self_repair = Keyword.get(opts, :from_self_repair, false)
+    from_db? = Keyword.get(opts, :from_db, false)
+    from_self_repair? = Keyword.get(opts, :from_self_repair, false)
 
-    do_load_transaction(tx, from_db, from_self_repair)
+    do_load_transaction(tx, from_db?, from_self_repair?)
   end
 
-  defp do_load_transaction(
-         tx = %Transaction{
-           address: address,
-           type: type,
-           data: %TransactionData{code: code}
-         },
-         _from_db,
-         _from_self_repair
-       )
-       when code != "" do
-    stop_contract(Transaction.previous_address(tx))
+  def do_load_transaction(tx, from_db?, from_self_repair?) do
+    %Transaction{
+      address: tx_address,
+      type: tx_type,
+      data: %TransactionData{code: code},
+      validation_stamp:  %ValidationStamp{recipients: recipients}
+    } = tx
 
+    cond do
+      code == "" && recipients == [] ->
```

Comment:
I would prefer pattern matching in the function signature which is more clear for me.